gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/http/connector/AbstractConnector.java

```diff
@@ -107,16 +108,21 @@ public ProxyConnection request(ProxyRequest proxyRequest) {
             final int port = url.getPort() != -1 ? url.getPort() :
                     protocol.charAt(protocol.length() - 1) == 's' ? SECURE_PORT : UNSECURE_PORT;
 
-            final String host = (port == UNSECURE_PORT || port == SECURE_PORT) ?
+            if (endpoint.getHttpClientOptions().getVersion() != ProtocolVersion.HTTP_2) {
+                final String host = (port == UNSECURE_PORT || port == SECURE_PORT) ?
                     url.getHost() : url.getHost() + ':' + port;
-
-            proxyRequest.headers().set(HttpHeaders.HOST, host);
+                proxyRequest.headers().set(HttpHeaders.HOST, host);
```

Comment:
I'm wondering if we should apply the same logic for both `Host` and `:authority` ?